src/librustc_typeck/check/method/probe.rs

```diff
@@ -128,7 +133,7 @@ pub enum PickKind<'tcx> {
                     ty::PolyTraitRef<'tcx>),
 }
 
-pub type PickResult<'tcx> = Result<Pick<'tcx>, MethodError<'tcx>>;
+pub type PickResult<'tcx> = Result<Vec<Pick<'tcx>>, MethodError<'tcx>>;
```

Comment:
Yes, and like I said, I didn't find a way to do it without completely splitting the code.